go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -367,6 +368,9 @@ func (ts *txThrottlerState) healthChecksProcessorFactory(topoServer *topo.Server
 			cellsUpdateTicker := time.NewTicker(topoCellsRefreshInterval)
 			for {
 				select {
+				case <-ctx.Done():
+					cellsUpdateTicker.Stop()
```

Comment:
Should instead `defer cellsUpdateTicker.Stop()` immediately after `cellsUpdateTicker := time.NewTicker(topoCellsRefreshInterval)`

Reply:
@shlomi-noach good point, I'll update 👍 

On `. healthChecksProcessorFactory()`: I was perhaps too clever - I avoided 2 x goroutines and a mutex by putting the ticker in the same loop as the health check processor

What are your thoughts on it? The other option I see is a dedicated goroutine with the cells update ticker and a mutex to synchronize

Reply:
Oh I'm in favor of a `select` based approach as it avoid the issue of protecting the list of cells via mutex. But you have to be prepared for "hiccups": if the code that reads ells from the topo runs too long, that in turn affects `<-ts.healthCheckChan:`. If you're OK with that then the design is good.

Reply:
> If you're OK with that then the design is good.

@shlomi-noach Good point, let's go with this for now and optimize later if needed 👍